icu4c/source/i18n/decimfmt.cpp

```diff
@@ -1152,6 +1442,9 @@ void DecimalFormat::touch(UErrorCode& status) {
                     NumberPropertyMapper::create(
```

Comment:
Idea: instead of using fields->formatter.adoptInstead(), do an std::move into the old object.  Then we don't have to actually allocate any new memory and can keep error states purely in the constructor.

Reply:
I was thinking of maybe using placement new, but `std::move` would likely be easier.
Unfortunately, `NumberPropertyMapper::create` can also fail due it internally trying to allocate memory as well, so using either placement new or `std::move` wouldn't be sufficient to avoid the OOM problems. :(

Reply:
Thanks for the idea!
Unfortunately, we can't always std::move a failed LocalizedNumberFormatter, as there can be the case where either `fields` or `fields->formatter` failed to allocate -- so we'd have nowhere to move it to.
However, if the allocation was successful, then we can indeed use std::move, which saves us from creating a new heap allocated LocalizedNumberFormatter every time `touch` is called. (Which given that touch is called fairly often would hopefully improve the performance somewhat.)